src/Microsoft.AspNet.Mvc.Abstractions/ControllerContext.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.Formatters;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.ModelBinding.Validation;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ControllerContext : ActionContext
```

Comment:
Two thoughts.

I considered adding a `new` property here for `ControllerActionDescriptor` so you don't need to cast if you want to see things like the type and methodinfo.

Doing that would pull this out of abstractions, and into core. Which seems like a good idea anyway. It's a detail of how controller + its invoker work.